pkg/controllers/job/job_controller.go

```diff
@@ -312,11 +321,15 @@ func (cc *Controller) processNextReq(count uint32) bool {
 	}
 
 	if err := st.Execute(action); err != nil {
-		glog.Errorf("Failed to handle Job <%s/%s>: %v",
-			jobInfo.Job.Namespace, jobInfo.Job.Name, err)
-		// If any error, requeue it.
-		queue.AddRateLimited(req)
-		return true
+		if queue.NumRequeues(req) < maxRetries {
+			glog.V(2).Infof("Failed to handle Job <%s/%s>: %v",
+				jobInfo.Job.Namespace, jobInfo.Job.Name, err)
+			// If any error, requeue it.
+			queue.AddRateLimited(req)
```

Comment:
We should update job status and record events accordingly.

Reply:
job status and event inside of the Execute function should be enough for this.

Reply:
what's kind of event in Execute? How user identify whether it's done?